### PR TITLE
Make it build with ghc-9.6.1

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -4,3 +4,13 @@ index-state: 2023-01-11T00:00:00Z
...
+
+allow-newer:
+  , *:base
+  , *:ghc-prim
+  , *:template-haskell
+  , hedgehog:*
+  , protolude:binary
+  , protolude:bytestring
+  , protolude:text
+  , th-compat:*
```
Comment (angerman) [inline cabal.project:16]: I believe we'd want to guard this being some 9.6 compat cabal.project.local for now?
We do something similar with custom cabal.project.locals in cardano-node.

Comment (erikd) [reply cabal.project]: Done!

**Review by angerman — APPROVED** [review]
LGTM

File: .github/workflows/haskell.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.2.5"]
+        ghc: ["8.10.7", "9.2.7", "9.6.1"]
```
Comment (Jimbo4350) [inline .github/workflows/haskell.yml:16]: Do we need to build all three of these?

Comment (erikd) [reply .github/workflows/haskell.yml:16]: For now we do.

As soon as the performance of the `ghc-9.2` build has be validated by QA we can drop `ghc-8.10`.

**Pull request merged**